### DISCUSSION TITLE
pod volume backup failed: error getting volume path on host: expected one matching path, got 0

### DISCUSSION THREAD

**Discussion by jbmassicotte** [vmware-tanzu/velero#3253] [Community support Q&A] [answered]
**UPDATE**: I modified this post given some recent observations.

After installing velero  version 1.5.1 with Azure plugin 1.1.0 on an Azure kubernetes cluster, and starting a schedule of backups, the restic portion of the first backup fails with this error:

> time="2020-11-17T17:32:43Z" level=error msg="Error backing up item" backup=velero/dci8hourbackups-20201117173204 error="pod volume backup failed: error getting volume path on host: expected one matching path, got 0" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/backupper.go:179" error.function="github.com/vmware-tanzu/velero/pkg/restic.(*backupper).BackupPodVolumes" logSource="pkg/backup/backup.go:455" name=dci-webservices-0

All subsequent backups have however consistently completed successfully. I was able to observe that exact behavior on 2 distinct systems.

Given those new facts this issue has dropped in priority for us, but we sure would like to understand what is going on to make sure we wont get bit in the future by some other manifestation of this problem.

I am mentioning here 2 posts that have reported the same error message. I read them but do not think the suggested explanations apply given the subsequent backups have succeeded in our case:

- https://github.com/konveyor/mig-controller/issues/207
- https://github.com/vmware-tanzu/velero/issues/1451

Any input would be greatly appreciated. Thanks in advance.

----

Note I first created this same issue in github project konveyor/mig-controller (konveyor/mig-controller/issues/800) given the issue 207 was in that project but decided I would have more luck recreating it in the velero project. Sorry for the noise.

**Comment by dsu-igeek** [comment, marked as answer]
It looks like mig-controller is handling the installation of Velero.  Please try installing Velero following our procedures (https://velero.io/docs/v1.5/basic-install/) and see if the problem happens.  We can start with figuring out if mig-controller is installing correctly. 

The output of the following commands will help us better understand what's going on:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`
